### ISSUE TITLE
`expr + "$x"` is not reported as non-POSIX

### ISSUE THREAD

**Issue by cD1rtX3** [koalaman/shellcheck#3358]
#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh
expr + "$1" : 'special'
```

#### Here's what shellcheck currently says:
Nothing.

#### Here's what I wanted or expected to see:
Something like “`expr + EXPRESSION` is undefined in POSIX sh”.
